app/notifiers/publishing_api_notifier.rb

```diff
@@ -1,4 +1,14 @@
+require 'sidekiq'
+
 class PublishingAPINotifier
+  def self.batch_send_to_publishing_api(tag)
+    if tag.child?
+      parent_with_children(tag).each { |item| queue_publishing_change(item.id) }
+    else
+      full_hierarchy.each { |item| queue_publishing_change(item.id) }
+    end
```

Comment:
This feels like knowledge that shouldn't be in this class.  I think this would be better refactored to be something like:
```ruby
def self.send_to_publishing_api(tag)
  send_to_publishing_api(tag)
  tag.dependent_tags.each { |dependent_tag| queue_publishing_change(dependent_tag.id) }
end
```
The existing `send_to_publishing_api` method can become a private method, and be renamed to something like `send_single_tag_to_publishing_api`.

That way, the knowledge of which tags depend on each other can be kept within the models, and this class is solely responsible for actually sending to the publishing-api.


This approach has the advantage that sending the tag itself remains synchronous, meaning that any errors will be reported to the user, and additionally, sending of dependent tags will only be queued if sending this tag succeeds.